scripts/src/modules/search/search-long-filters.js

```diff
@@ -0,0 +1,96 @@
+export default function () {
+    const $parentForm = document.querySelector('[data-id="long-filter-form"]');
+
+    const $longFiltersContainer = document.querySelector('[data-id="long-filters-container"]');
+
+    const $longFiltersList = document.querySelector('[data-id="long-filters-list"]');
+    const $longFiltersListItems = document.querySelectorAll('[data-class="long-filters-list-item"]');
+
+    const $longFiltersLegend = document.querySelector('[data-id="long-filters-legend"]');
+    const $longFiltersFieldset = document.querySelector('[data-id="long-filters-fieldset"]');
+
+    if(!$parentForm || !$longFiltersContainer || !$longFiltersListItems || !$longFiltersList || !$longFiltersLegend || !$longFiltersFieldset) {
+        return;
+    }
+
+    const longFiltersArray = [].slice.call($longFiltersListItems); // IE11 equivalent of doing [...$longFiltersListItems]
+
+    const $searchDiv = document.createElement('div');
+    $searchDiv.setAttribute('class', 'long-filters__search');
+    const $searchBox = document.createElement('input');
+    $searchBox.setAttribute('class', 'long-filters__search-box');
+    $searchBox.setAttribute('type', 'search');
+    const searchId = 'long-filters-search-box';
+    $searchBox.id = searchId;
+    const $searchLabel = document.createElement('label');
+    $searchLabel.innerText = "Narrow down filters";
+    $searchLabel.setAttribute('for', searchId);
+    $searchLabel.setAttribute('class', 'long-filters__search-label');
+
+    let $searchHelperText = document.createElement('p');
+    $searchHelperText.innerText = 'Filters that contain the text you enter will be displayed. Filters you have checked will always be shown.';
+    $searchHelperText.id = 'long-filters-helper-text';
+    $searchBox.setAttribute('aria-describedby', 'long-filters-helper-text');
+
+    $searchDiv.appendChild($searchLabel);
+    $searchDiv.appendChild($searchBox);
+    $searchDiv.appendChild($searchHelperText);
+
+    const $filterCount = document.createElement('p');
+    $filterCount.id = 'long-filters-count';
+    $filterCount.setAttribute('class', 'long-filters__count');
+    $longFiltersFieldset.setAttribute('aria-describedby', 'long-filters-count');
+
+    const totalFiltersLength = longFiltersArray.length;
+    let longFiltersCountText = `Showing ${totalFiltersLength} out of ${totalFiltersLength} filters`;
+    $filterCount.innerText = longFiltersCountText;
+
+    $longFiltersFieldset.insertBefore($filterCount, $longFiltersFieldset.childNodes[2]);
+
+    $longFiltersContainer.insertBefore($searchDiv, $longFiltersContainer.childNodes[0]);
+
+    const handleSearch = function(e) {
+        e.preventDefault();
```

Comment:
Could this be simplified to:

`if(e.key === “Enter”) { 
      e.preventDefault(); 
  }`
  
  And then replace line 94 with something like `$searchBox.addEventListener('keypress', handleSearch);`?
  
Then I think `disableSearchSubmit` wouldn't be needed. Just my opinion though so if it's preferable to keep it as it is then that's cool with me.

Reply:
Actually, this might not be a good idea. Just realised that this would result in calling the `handleSearch` function twice every time a key is pressed.

Reply:
Thanks @JamesWChan  - I also think the `preventDefault` has to be on the `$parentForm` element, as with forms it will consider any Enter keypress while inside the form as a submit, so it can't be on the `$searchBox` itself. The `id` checking is so that the submit buttons and other form elements can still trigger a submission by pressing enter.